lib/publiccloud/azure.pm

```diff
@@ -69,12 +69,11 @@ sub vault_create_credentials {
     my ($self) = @_;
 
     record_info('INFO', 'Get credentials from VAULT server.');
-    my $res = $self->vault_api('/v1/' . get_var('PUBLIC_CLOUD_VAULT_NAMESPACE', '') . '/azure/creds/openqa-role', method => 'get');
-    $self->vault_lease_id($res->{lease_id});
-    $self->key_id($res->{data}->{client_id});
-    $self->key_secret($res->{data}->{client_secret});
+    my $data = $self->vault_get_secrets('/v1/' . get_var('PUBLIC_CLOUD_VAULT_NAMESPACE', '') . '/azure/creds/openqa-role');
```

Comment:
```suggestion
    my $data = $self->vault_get_secrets('/azure/creds/openqa-role');
```

can we move same part inside function ?

Reply:
We can, as we now agreed that each our credentials in such a subfolder. I will change it.